File: backend/src/main/java/kr/co/programmers/collabond/api/apply/domain/dto/ApplyPostDto.java
package kr.co.programmers.collabond.api.apply.domain.dto;

import kr.co.programmers.collabond.api.apply.domain.ApplyPost;
import kr.co.programmers.collabond.api.apply.interfaces.ApplyPostMapper;
import kr.co.programmers.collabond.api.attachment.domain.dto.AttachmentDto;
import kr.co.programmers.collabond.api.profile.domain.dto.ProfileDto;
import kr.co.programmers.collabond.api.recruit.domain.dto.RecruitPostDto;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@NoArgsConstructor
public class ApplyPostDto {

    private Long id;

    private RecruitPostDto recruitPost;

    private ProfileDto profile;

    private String content;

    private String status;

    private List<AttachmentDto> attachmentFiles;

    private LocalDateTime createdAt;

    @Builder
    private ApplyPostDto(Long id,
                         RecruitPostDto recruitPost,
                         ProfileDto profile,
                         String content,
                         String status,
                         List<AttachmentDto> attachmentFiles,
                         LocalDateTime createdAt) {

        this.id = id;
        this.recruitPost = recruitPost;
        this.profile = profile;
        this.content = content;
        this.status = status;
        this.attachmentFiles = attachmentFiles;
        this.createdAt = createdAt;
    }

    public ApplyPostDto(
            ApplyPost applyPost
    ) {
        ApplyPostDto a = ApplyPostMapper.toDto(applyPost);
        this.id = a.getId();
        this.recruitPost = a.getRecruitPost();
        this.profile = a.getProfile();
        this.content = a.getContent();
        this.status = a.getStatus();
        this.attachmentFiles = a.getAttachmentFiles();
        this.createdAt = a.getCreatedAt();
    }
}

File: backend/src/main/java/kr/co/programmers/collabond/api/apply/domain/dto/ApplyPostDto.kt
package kr.co.programmers.collabond.api.apply.domain.dto

import kr.co.programmers.collabond.api.apply.domain.ApplyPost
import kr.co.programmers.collabond.api.apply.interfaces.ApplyPostMapper
import kr.co.programmers.collabond.api.attachment.domain.dto.AttachmentDto
import kr.co.programmers.collabond.api.profile.domain.dto.ProfileDto
import kr.co.programmers.collabond.api.recruit.domain.dto.RecruitPostDto
import java.time.LocalDateTime

data class ApplyPostDto(
    var id: Long,
    var recruitPost: RecruitPostDto?,
    var profile: ProfileDto?,
    var content: String,
    var status: String,
    var attachmentFiles: MutableList<AttachmentDto>?,
    var createdAt: LocalDateTime,
) {

    constructor(
        applyPost: ApplyPost
    ) : this(
        applyPost.id,
        null,
        null,
        applyPost.content,
        applyPost.status.toString(),
        null,
        applyPost.createdAt
    ) {
        val a = ApplyPostMapper.toDto(applyPost)
        this.recruitPost = a.recruitPost
        this.profile = a.profile
        this.attachmentFiles = a.attachmentFiles
    }
}
